src/Components/News/NewsCreateEdit.js

```diff
@@ -0,0 +1,231 @@
+                }
+            }).catch(err => {
+                alert(err)
+            })
+        }
+        if (id) {
+            axios.put(urlEditarNovedad, {
+                name: initialValues.title,
+                image: initialValues.image,
+                content: datosSinEtiquetas,
+                category_id: initialValues.category
+            }).then(res => {
+                if (res.status === 200) {
+                    alert("news updated successfulli");
+                    return setInitialValues(datosForm)
+                }
+            }).catch(err => {
+                alert(err)
+            })
+        }
+        e.preventDefault();
+        console.log(initialValues);
+    }
+
+    useEffect(() => {
+
+        const DatosEditNew = async () => {
+            const DataInicialCategoria = await axios.get(urlCategoria),
+                categoriaData = await DataInicialCategoria.data.data;
+            setcategorias(categoriaData)
+
+            if (id) {
+                const datosIniciales = await axios.get(urlEditarNovedad),
+                    EditNewData = datosIniciales.data.data,
+                    { name, content } = await EditNewData
+                if (name) setInitialValues({ ...initialValues, title: name })
+                if (content) setInitialValues({ ...initialValues, content: content })
+            }
+
+        }
+
+        DatosEditNew()
+    }, [urlCategoria])
+
+    return (
+        
+        <>
+            <form className="form-container" onSubmit={handleSubmit}>
+                <label htmlFor="title" className='label-Title-New-News'>
+                    <h2 className="titulo-Titulo-New-News">Title</h2>
+                    <input
+                        required
+                        className="input-field"
+                        type="text"
+                        name="title"
+                        value={initialValues.title || ''}
+                        onChange={handleChange}
+                    ></input>
+                </label>
+                <h2 className="titulo-Content-New-News">Description</h2>
+                <CKEditor 
+                    config={editorConfiguration}
+                    editor={ClassicEditor}
+                    data={initialValues.content}
+                    onChange={contentChange}
+                />
+                <h2 className="categorias-New-News">Category</h2>
+                <select
+                    required
+                    className="select-field"
+                    name="category"
+                    value={initialValues.category || ''}
+                    onChange={handleChange}
+                >
+                    <option value="">Select category</option>
+                    {
+                        categorias?.map((e) => {
+                            return <option key={e.id} value={e.name}>{e.name}</option>
+                        })
+                    }
```

Comment:
move this code to a function out of the view
```
const showCategoryOptions = () => categories?.map( category => (<option key={category.id} value={category.name}>{category.name}</option>) );
```
and then use it here:
```
<option value="">Select category</option>
{ showCategoryOptions() }
```

Reply:
resuelto